docs/blueprints/deletion-performance.md

```diff
@@ -0,0 +1,188 @@
+- Storage nodes will have more garbage if we use a probabilistic approach.
+
+
+### (2) Satellite delete pieces of the storage nodes reliably
+
+Uplink will communicate with the satellite as it currently does.
+
+The satellite will take care of communicating with the storage nodes for deleting the pieces using RPC.
+
+Advantages:
+
+- Uplink deletion operation will be independent of the size of the file, guaranteeing always being responsive.
+- It doesn't present a risk of leaving garbage in the storage nodes when deletion operation is interrupted.
+- In general, the storage nodes will have less garbage because of deletions.
+
+Disadvantages:
+
+- The satellite requires a new chore to delete the pieces of the storage nodes. The increment of network traffic, computation, and data to track the segments to delete will increase the running costs.
+- The satellite will have another component incrementing the cost of the operation as monitoring, replication, etc.
+
+
+### (3) Satellite delete pieces of the storage nodes unreliably
+
+Uplink will communicate with the satellite as it currently does.
+
+The satellite will take care of communicating with the storage nodes for deleting the pieces using a connectionless protocol like UDP.
+
+Advantages:
+
+- Uplink deletion operation will be independent of the size of the file, guaranteeing always being responsive.
+- It doesn't present a risk of leaving garbage in the storage nodes when deletion operation is interrupted.
+
+Disadvantages:
+
+- The satellite requires a new chore to delete the pieces of the storage nodes. The increment of network traffic, computation, and data to track the segments to delete will increase the running costs.
+- The satellite will have another component incrementing the cost of the operation as monitoring, replication, etc.
+
+### Conclusion
+
+The alternative approach (1):
+
+- It is similar to the current mechanism but with some improvements towards the goals.
+- It doesn't add more load to the satellite, but we cannot trust in the uplink in deleting the pieces or informing the non-deleted ones.
+
+The alternative approaches (2) and (3) are similar.
+
+Approach (2) has the advantage of guaranteeing less garbage left on the storage nodes at the expense of requiring more network traffic.
+
+Approach (3) requires less network traffic, but it may not require less computation considering that we may need to encrypt the data sent through UDP.
+
+Approach (3) may get rid of the satellite garbage faster.
+
+Both approaches, (2) and (3), present the problem of increasing garbage on the satellite side.
+
+Taking one of these approaches will require a study on how to keep the less amount of garbage as possible.
+
+## Implementation
+
+### (1) Storage nodes:
+
+1. Adapt protocol buffers definitions for delete operation.
+       Satellite doesn't have to send order limits; it only has to send piece ID.
+1. Adapt the delete endpoint to receive requests from the satellite.
```

Comment:
this part is where i thought there would be a new endpoint so we don't need to worry about protobuf backwards compatibility. when the satellite issues a delete to the storage nodes, it goes to a new storage node endpoint, like SatelliteDelete or something

Reply:
Oh, yes, we are doing that, I updated the ticked after the last arch meeting but not the document.
I wrote this after the discussion in our team but before we discussed it in the last arch meeting.

I'm merging it and sending a new PR with this subtle change.

Reply:
https://github.com/storj/storj/pull/3601